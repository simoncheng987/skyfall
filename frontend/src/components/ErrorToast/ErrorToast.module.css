.container {
    border-radius: 15px;
    display: inline-block;
    background-color: #D42F2F;
    border: 2px solid #000000;
    box-shadow: 0px 4px 0px #1C1C1C;
}

.flexbox {
    padding: 30px 30px;
    display: flex;
    flex-direction: row;
    align-items: center;
}

.icon {
    filter: drop-shadow(0px 4px 0px #1C1C1C);
    flex-shrink: 0;
}

.text {
    color: white;
    font-family: "Poppins";
    font-weight: 500;
    padding-left: 15px;
    font-size: 18px;
}