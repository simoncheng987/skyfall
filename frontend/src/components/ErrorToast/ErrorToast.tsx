import React from 'react';
import { ExclamationCircleIcon } from '@heroicons/react/solid';
import styles from './ErrorToast.module.css';

interface ErrorToastProps {
  text: string;
  className?: string;
}

export default function ErrorToast({ text, className }: ErrorToastProps) {
  return (
    <div className={`${className} ${styles.container}`}>
      <div className={styles.flexbox}>
        <ExclamationCircleIcon className={styles.icon} width={30} color="white" />
        <p className={styles.text}>{text}</p>
      </div>
    </div>
  );
}

ErrorToast.defaultProps = {
  className: '',
};
