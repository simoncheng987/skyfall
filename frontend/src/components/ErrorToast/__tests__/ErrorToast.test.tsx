import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import ErrorToast from '../ErrorToast';

describe('ErrorToast', () => {
  const className = 'testClassName';
  const errorMessage = 'ErrorToast Message';

  it('renders the error text prop', () => {
    render(<ErrorToast text={errorMessage} />);
    expect(screen.getByText(errorMessage)).toBeInTheDocument();
  });

  it('matches snapshot with no className prop', () => {
    const tree = renderer.create(<ErrorToast text={errorMessage} />).toTree();
    expect(tree).toMatchSnapshot();
  });

  it('matches snapshot with className prop', () => {
    const tree = renderer.create(<ErrorToast className={className} text={errorMessage} />).toTree();
    expect(tree).toMatchSnapshot();
  });
});
