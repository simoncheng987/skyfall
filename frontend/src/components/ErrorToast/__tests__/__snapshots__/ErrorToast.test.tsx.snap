// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorToast matches snapshot with className prop 1`] = `
Object {
  "instance": null,
  "nodeType": "component",
  "props": Object {
    "className": "testClassName",
    "text": "ErrorToast Message",
  },
  "rendered": Object {
    "instance": null,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="flexbox"
      >
        <ForwardRef(ExclamationCircleIcon)
          className="icon"
          color="white"
          width={30}
        />
        <p
          className="text"
        >
          ErrorToast Message
        </p>
      </div>,
      "className": "testClassName container",
    },
    "rendered": Array [
      Object {
        "instance": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <ForwardRef(ExclamationCircleIcon)
              className="icon"
              color="white"
              width={30}
            />,
            <p
              className="text"
            >
              ErrorToast Message
            </p>,
          ],
          "className": "flexbox",
        },
        "rendered": Array [
          Object {
            "instance": null,
            "nodeType": "host",
            "props": Object {
              "aria-hidden": "true",
              "children": <path
                clipRule="evenodd"
                d="M18 10a8 8 0 11-16 0 8 8 0 0116 0zm-7 4a1 1 0 11-2 0 1 1 0 012 0zm-1-9a1 1 0 00-1 1v4a1 1 0 102 0V6a1 1 0 00-1-1z"
                fillRule="evenodd"
              />,
              "className": "icon",
              "color": "white",
              "fill": "currentColor",
              "viewBox": "0 0 20 20",
              "width": 30,
              "xmlns": "http://www.w3.org/2000/svg",
            },
            "rendered": Array [
              Object {
                "instance": null,
                "nodeType": "host",
                "props": Object {
                  "clipRule": "evenodd",
                  "d": "M18 10a8 8 0 11-16 0 8 8 0 0116 0zm-7 4a1 1 0 11-2 0 1 1 0 012 0zm-1-9a1 1 0 00-1 1v4a1 1 0 102 0V6a1 1 0 00-1-1z",
                  "fillRule": "evenodd",
                },
                "rendered": Array [],
                "type": "path",
              },
            ],
            "type": "svg",
          },
          Object {
            "instance": null,
            "nodeType": "host",
            "props": Object {
              "children": "ErrorToast Message",
              "className": "text",
            },
            "rendered": Array [
              "ErrorToast Message",
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  "type": [Function],
}
`;

exports[`ErrorToast matches snapshot with no className prop 1`] = `
Object {
  "instance": null,
  "nodeType": "component",
  "props": Object {
    "className": "",
    "text": "ErrorToast Message",
  },
  "rendered": Object {
    "instance": null,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="flexbox"
      >
        <ForwardRef(ExclamationCircleIcon)
          className="icon"
          color="white"
          width={30}
        />
        <p
          className="text"
        >
          ErrorToast Message
        </p>
      </div>,
      "className": " container",
    },
    "rendered": Array [
      Object {
        "instance": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <ForwardRef(ExclamationCircleIcon)
              className="icon"
              color="white"
              width={30}
            />,
            <p
              className="text"
            >
              ErrorToast Message
            </p>,
          ],
          "className": "flexbox",
        },
        "rendered": Array [
          Object {
            "instance": null,
            "nodeType": "host",
            "props": Object {
              "aria-hidden": "true",
              "children": <path
                clipRule="evenodd"
                d="M18 10a8 8 0 11-16 0 8 8 0 0116 0zm-7 4a1 1 0 11-2 0 1 1 0 012 0zm-1-9a1 1 0 00-1 1v4a1 1 0 102 0V6a1 1 0 00-1-1z"
                fillRule="evenodd"
              />,
              "className": "icon",
              "color": "white",
              "fill": "currentColor",
              "viewBox": "0 0 20 20",
              "width": 30,
              "xmlns": "http://www.w3.org/2000/svg",
            },
            "rendered": Array [
              Object {
                "instance": null,
                "nodeType": "host",
                "props": Object {
                  "clipRule": "evenodd",
                  "d": "M18 10a8 8 0 11-16 0 8 8 0 0116 0zm-7 4a1 1 0 11-2 0 1 1 0 012 0zm-1-9a1 1 0 00-1 1v4a1 1 0 102 0V6a1 1 0 00-1-1z",
                  "fillRule": "evenodd",
                },
                "rendered": Array [],
                "type": "path",
              },
            ],
            "type": "svg",
          },
          Object {
            "instance": null,
            "nodeType": "host",
            "props": Object {
              "children": "ErrorToast Message",
              "className": "text",
            },
            "rendered": Array [
              "ErrorToast Message",
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  "type": [Function],
}
`;
